addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/EnvFileScanRule.java

```diff
@@ -38,8 +49,31 @@ public int getId() {
 
     @Override
     public boolean isFalsePositive(HttpMessage msg) {
+        // We need the response header and the body
+        HttpResponseHeader responseHeader = msg.getResponseHeader();
+        String contentType = responseHeader.getNormalisedContentTypeValue();
         String responseBody = msg.getResponseBody().toString();
-        // It likely is a FP if the response contains neither a comment nor assignment
-        return !responseBody.contains("#") && !responseBody.contains("=");
+
+        // If we don't get a response header, the status code isn't 200 or content length is > 500,
+        // prolly not a .env
+        if (responseHeader.isEmpty()
+                || !HttpStatusCode.isSuccess(200)
+                || responseBody.length() > 500) return true;
+
+        // Apache will not set a content-type header and NGINX sets
+        // content-type=application/octet-stream
+        if (contentType == null || contentType.equals("application/octet-stream")) {
+            // Our reg_ex patterns we are after
+            Pattern comment_pattern = Pattern.compile("#\\s{0,10}\\w+");
```

Comment:
Move these to the top so we just compile them once and not on every request.

Reply:
Not that it matters since they will be removed, but it's better to follow Java naming conventions.

Reply:
@thc202 do you mean commentPattern for that variable name, if so I can fix and sorry my python brain is always working overtime.

Reply:
Now it should be called `COMMENT_PATTERN` (the string can be passed directly to compile).

Reply:
Yes

Reply:
The other variables still need to be updated though.